products/green_fast_track/macros/clip_to_geom.sql

```diff
@@ -0,0 +1,9 @@
+{% macro clip_to_geom(left, right=ref("stg__nyc_boundary"), left_by="geom", right_by="geom") %}
+
+    SELECT
+        {{ dbt_utils.star(from=left, except=[left_by]) }},
+        ST_INTERSECTION({{ left }}.{{ left_by }}, {{ right }}.{{ right_by }}) AS geom
+    FROM {{ left }}
```

Comment:
might be worth adding ability to get some sort of identifier from `right`. For now, stg__nyc_boundary only has one row so it's not relevant. But you could imagine that if you want to clip to boroughs, each row returned should have a boro_id attached

Reply:
definitely would be a nice feature, but I think my brain processes the singular geom in `clip_to_geom` as involving a one row in the right table anyway